### ISSUE TITLE
Error training multiple epochs

### ISSUE THREAD

**Issue by mortont** [ibab/tensorflow-wavenet#66]
Training on a truncated sample of the VCTK corpus causes it to hang with no error if it hits the end of the corpus before the number of steps is complete. It looks like this is due to `iterate_through_vctk` not resetting once it hits the end. It looks like this is also happening in #65 on the full corpus, since it's stalling at 44256 steps, which is close to the samples in the VCTK corpus (109 speakers x ~400 samples/speaker).

I've tested this with an expanded file list (to iterate through multiple epochs) and it no longer stops at the end of the corpus.

I suggest we implement a `tf.train.string_input_producer` to iterate through epochs. That way we can also shuffle input, as has been mentioned in the discussion in #47.

**Comment by mortont** [comment]
Looks like I was on an older commit, feel free to close this error. It may be still be a good idea to look at using `tf.train.string_input_producer` though.

**Comment by lemonzi** [comment]
I agree, at some point we could switch to TF's file globber and input producers. There is also TensorFlow.slim's Dataset and DataProducer classes that we could make use of. Feel free to send a PR for that, or open a more specific issue.
